pkg/ccl/backupccl/restore_online_test.go

```diff
@@ -82,6 +81,75 @@ func TestOnlineRestoreBasic(t *testing.T) {
 	rSQLDB.CheckQueryResults(t, createStmt, createStmtRes)
 }
 
+func TestOnlineRestoreStatementResult(t *testing.T) {
+	defer leaktest.AfterTest(t)()
+	defer log.Scope(t).Close(t)
+	defer nodelocal.ReplaceNodeLocalForTesting(t.TempDir())()
+
+	const numAccounts = 1
+	_, sqlDB, _, cleanupFn := backupRestoreTestSetupWithParams(
+		t,
+		singleNode,
+		numAccounts,
+		InitManualReplication,
+		base.TestClusterArgs{
+			ServerArgs: base.TestServerArgs{
+				DefaultTestTenant: base.TestIsSpecificToStorageLayerAndNeedsASystemTenant,
+			},
+		},
+	)
+	defer cleanupFn()
+
+	sqlDB.ExecMultiple(
+		t,
+		"USE data",
+		"CREATE TABLE foo (x INT PRIMARY KEY, y INT)",
+		"INSERT INTO foo VALUES (1, 2)",
+	)
+	sqlDB.Exec(t, "BACKUP DATABASE data INTO $1", localFoo)
+
+	rows := sqlDB.Query(
+		t,
+		"RESTORE DATABASE data FROM LATEST IN $1 WITH OPTIONS (new_db_name='data2', experimental deferred copy)",
+		localFoo,
+	)
+	columns, err := rows.Columns()
+	if err != nil {
+		t.Fatal(err)
+	}
+	if a, e := columns, []string{
+		"job_id", "tables", "approx_rows", "approx_bytes", "background_download_job_id",
+	}; !reflect.DeepEqual(e, a) {
+		t.Fatalf("unexpected columns:\n%s", strings.Join(pretty.Diff(e, a), "\n"))
+	}
+	if !rows.Next() {
+		if err := rows.Err(); err != nil {
+			t.Fatal(err)
+		}
+		t.Fatal("zero rows in result")
+	}
+
+	var id, expectedID, tables, approxRows, approxBytes, downloadJobID int64
+	require.NoError(t, rows.Scan(
+		&id, &tables, &approxRows, &approxBytes, &downloadJobID,
+	))
+	sqlDB.QueryRow(t,
+		`SELECT job_id FROM crdb_internal.jobs WHERE job_id = $1`, id,
+	).Scan(
+		&expectedID,
+	)
+
+	require.Equal(t, expectedID, id, "result does not match system.jobs")
+	require.NotZero(t, downloadJobID, "expected a background download job ID")
```

Comment:
Since it seems like we are tentatively deciding to stick with `expectedID + 1`, I figured the test should be written to just ensure that there is a background job ID set instead of any value in particular.

Do you think it's better to set it to `expectedID + 1` and if we change the background job ID generation in the future, to come back and update the test?

Reply:
> Do you think it's better to set it to expectedID + 1 and if we change the background job ID generation in the future, to come back and update the test?
Yeah exactly. If that patch ever comes, this test will fail and the patch will have to modify this test, which is totally fine.

Reply:
Ah sounds good, so the same logic applies to the tables comment below then